src/index.ts

```diff
@@ -11,22 +11,58 @@ require('dotenv').config({
 import logProcessError from 'log-process-errors';
 import { createLogger } from './logging';
 import { getConfig, KeeperConfig } from './config';
-import { getDefaultProvider, utils, Wallet } from 'ethers';
+import { utils, Wallet, providers } from 'ethers';
 import { getPythDetails, getSynthetixPerpsContracts } from './utils';
 import { Distributor } from './distributor';
 import { LiquidationKeeper } from './keepers/liquidation';
 import { DelayedOrdersKeeper } from './keepers/delayedOrders';
 import { DelayedOffchainOrdersKeeper } from './keepers/delayedOffchainOrders';
 import { Metric, Metrics } from './metrics';
+import { Network } from './typed';
 
-export async function run(config: KeeperConfig) {
-  const logger = createLogger('Application');
-  const metrics = Metrics.create(config.isMetricsEnabled, config.network, config.aws);
+const logger = createLogger('Application');
+
+// 750ms.
+//
+// Waits `n` ms before executing the same request to the next provider ordered by priority.
+export const PROVIDER_STALL_TIMEOUT = 750;
+export const PROVIDER_DEFAULT_WEIGHT = 1;
+
+export const getProvider = async (
+  config: KeeperConfig['providerApiKeys'],
+  network: Network
+): Promise<providers.BaseProvider> => {
+  // Infura has the highest priority (indicated by the lowest priority number).
+  const providersConfig: providers.FallbackProviderConfig[] = [
+    {
+      provider: new providers.InfuraProvider(network, config.infura),
+      priority: 10,
+      stallTimeout: PROVIDER_STALL_TIMEOUT,
+      weight: PROVIDER_DEFAULT_WEIGHT,
+    },
+  ];
+  if (config.alchemy) {
+    logger.info('Alchemy API key provided. Adding as fallback provider');
+    providersConfig.push({
+      provider: new providers.AlchemyProvider(network, config.alchemy),
+      priority: 20,
+      stallTimeout: PROVIDER_STALL_TIMEOUT,
+      weight: PROVIDER_DEFAULT_WEIGHT,
+    });
+  }
+  return new providers.FallbackProvider(providersConfig, 1);
```

Comment:
```suggestion
  return new providers.FallbackProvider(providersConfig);
```
Dont ask me why, but if you provide quorum 1 it will do requests to alchemy...